### PR TITLE
DIM-16: allow reading from multiple Terraform state files

### PR DESCRIPTION
**Pull request by blen2r**
### What does this change do?
It allows Deployfish to read values from multiple Terraform statefiles

### Why is this change needed (what problem does it solve)?
https://tweedinc.atlassian.net/browse/DIM-16

### How has this been tested?

- [ ] Includes automated tests for _all_ functionality
- [ ] Includes automated tests for some functionality
- [x] Manual testing performed

I have tested that the values could be read but I couldn't confirm that the deployment actually works. We should:
1. Merge this PR
2. Redeploy an app that uses deployfish without Terraform
3. Deploy Graylog/Fluentd as they only use one statefile (to confirm that previous behaviour is still ok)
4. Deploy Dimebag to make sure multiple statefiles work

### Screenshots (if appropriate):
N/A

### Types of changes

- [ ] Doc change or reformatting (no change to code behavior)
- [ ] Tests only
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Known Issues, Open Questions, Comments
See notes in testing section

### CODE REVIEW

File: deployfish/config.py [hunk 1]
```diff
@@ -196,7 +196,17 @@ def replace_terraform(self):
             if 'workspace' in self.__raw['terraform']:
                 self.__raw['terraform']['workspace'] = self.__raw['terraform']['workspace'].format(**replacers)
             else:
-                self.__raw['terraform']['statefile'] = self.__raw['terraform']['statefile'].format(**replacers)
+                if isinstance(self.__raw['terraform'], dict):
+                    try:
+                        self.__raw['terraform']['statefile'] = self.__raw['terraform']['statefile'].format(**replacers)
```
Comment (blen2r) [inline deployfish/config.py]: Given how the 'statefile' field is a string and never contains the keys found in 'replacers', I don't see how this was ever working. I ended up skipping that step on errors.

**Comment by blen2r** [comment]
Ready for review

**Comment by blen2r** [comment]
Will reopen with task dependencies

**Comment by blen2r** [comment]
No changes needed for migrations after all.

**Comment by butlertron** [comment]
FYI if you want to test deployfish locally, what I usually do is change my `pyproject.toml` to something like

```
deployfish = { path = "../deployfish/" }
```
and then poetry will use the local version instead of the master

**Comment by blen2r** [comment]
Oh yeah I do that already. The problem comes from the KMS restrictions, I can't fully test the entire process as I'm not running on AWS, I've emulated what I could but I can't guarantee it's going to work the same way until we actually try it.

**Pull request merged**